components/Currency/CurrencyPrice/CurrencyPrice.tsx

```diff
@@ -4,54 +4,54 @@ import styled, { css } from 'styled-components';
 
 import ChangePercent from 'components/ChangePercent';
 import { ContainerRowMixin } from 'components/layout/grid';
+import { NumericValue } from 'components/Text';
 import { CurrencyKey } from 'constants/currency';
 import { formatCurrency, FormatCurrencyOptions } from 'utils/formatters/number';
 
 type CurrencyPriceProps = {
-	currencyKey: CurrencyKey;
+	currencyKey?: CurrencyKey;
 	showCurrencyKey?: boolean;
 	price: WeiSource;
 	sign?: string;
 	change?: number;
 	conversionRate?: WeiSource;
 	formatOptions?: FormatCurrencyOptions;
 	truncate?: boolean;
-	side?: 'positive' | 'negative';
+	side?: 'positive' | 'negative' | 'preview';
+	colored?: boolean;
 };
 
 export const CurrencyPrice: FC<CurrencyPriceProps> = memo(
```

Comment:
Maybe one for another PR but feel this is something that could be refactored or re-organised at least to fit with our new component lib.

Reply:
True. I was trying to do that in this PR, but this component needs a major rework (especially with the new `NumericValue` stuff we have going now).